save-backend/src/main/kotlin/com/saveourtool/save/backend/repository/TimestampBasedFileSystemRepository.kt

```diff
@@ -53,21 +38,51 @@ class TimestampBasedFileSystemRepository(configProperties: ConfigProperties) {
         }
     }
 
-    private fun getStorageDir(fileInfo: FileInfo) = rootDir.resolve(fileInfo.uploadedMillis.toString())
-
-    private fun createStorageDir(fileInfo: FileInfo) = getStorageDir(fileInfo).createDirectory()
+    private fun getStorageDir(
+        fileInfo: FileInfo,
+        organizationName: String,
+        projectName: String,
+    ) = rootDir
+        .resolve(organizationName)
+        .resolve(projectName)
+        .resolve(fileInfo.uploadedMillis.toString())
+
+    private fun createStorageDir(
+        fileInfo: FileInfo,
+        organizationName: String,
+        projectName: String,
+    ) = getStorageDir(fileInfo, organizationName, projectName).createDirectory()
 
     /**
+     * @param organizationName
+     * @param projectName
      * @return list of files in [rootDir]
      */
-    fun getFilesList() = rootDir.listDirectoryEntries()
-        .filter { it.isDirectory() }
-        .flatMap { it.listDirectoryEntries() }
+    fun getFilesList(
+        organizationName: String,
+        projectName: String,
+    ): List<Path> {
+        rootDir
+            .resolve(organizationName)
+            .resolve(projectName).let { path ->
+                if (path.exists()) {
+                    return path.listDirectoryEntries()
+                        .filter { it.isDirectory() }
+                        .flatMap { it.listDirectoryEntries() }
+                }
+            }
+        return emptyList()
```

Comment:
```suggestion
            .resolve(projectName)
            .takeIf { it.exists()}
            ?.listDirectoryEntries()
            ?.filter { it.isDirectory() }
            ?.flatMap { it.listDirectoryEntries() }
            ?: emptyList()
```
WDYT about this? This way you won't need additional nesting

Reply:
Yes. Thank you! Fixed